### DISCUSSION TITLE
Using service role for Discord bot. Am I doing this right?

### DISCUSSION THREAD

**Discussion by esomwrld** [supabase/supabase#12553] [Questions]
Hey everyone,

First time user/builder with Supabase and have everything setup and working with my React app. I have a question relating to RLS policies.

My app lets users create messages on the front end, and then I have a bot that delivers messages to Discord. I have setup RLS for my users which was easy enough, so they can read/update/delete their own messages. This was easy with the authenticated role and works fine.

I have a bot that needs to access all tables I have to update users and pull data (messages). For the bot, I have it currently accessing the DB via the service role. 

This works fine. But as I am new to Supabase I want to make sure I am not overlooking something. All keys are stored in an ENV file and I pull from there. I assume this makes the fact I am using a service role for my bot secure?

If not, how am I best approaching this, so that my bot can read/update and delete data for all users in my tables?

Really appreciate any help, as I can't find any specific explanations of this use case online.

**Comment by Mandalorian007** [comment]
I am also curious on this subject. Ideally, if you have a website with google sign in, but also a discord bot that can work off the same data you would want the user to also auth via discord this way discord actions could be taken as a logged in user for supabase's RLS. However, I have not been able to find a good example or discussion on this. It would be really nice if this topic could be covered as part of the discord bot example: 

https://github.com/supabase/supabase/blob/master/examples/edge-functions/supabase/functions/discord-bot/index.ts
